apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,642 @@
+}) {
+  return html`
+    <form name="add-member-form" method="POST">
+      UIDs:
+      <input
+        type="text"
+        class="form-control"
+        placeholder="A@ex.com, B@ex.com"
+        name="add_member_uids"
+      />
+      <br />
+      <input type="hidden" name="__action" value="add_member" />
+      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${iRow}PopoverAdd').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-primary">Add</button>
+    </form>
+  `;
+}
+
+function formDeleteGroup({
+  row,
+  iRow,
+  resLocals,
+}: {
+  row: GroupUsersRow;
+  iRow: number;
+  resLocals: Record<string, any>;
+}) {
+  return html`
+    <form name="delete-group-form" method="POST">
+      <p>Are you sure you want to delete the group <strong>${row.name}</strong>?</p>
+      <input type="hidden" name="__action" value="delete_group" />
+      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${iRow}Popoverdeletegroup').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-danger">Delete</button>
+    </form>
+  `;
+}
+
+function formRemoveMembers({
+  row,
+  iRow,
+  resLocals,
+}: {
+  row: GroupUsersRow;
+  iRow: number;
+  resLocals: Record<string, any>;
+}) {
+  return html`
+    <form name="delete-member-form" method="POST">
+      <div class="form-group">
+        <label for="delete-member-form-${iRow}">UID:</label>
+        <select class="form-control" name="user_id" id="delete-member-form-${iRow}">
+          ${row.users.map((user) => {
+            return html` <option value="${user.user_id}">${user.uid}</option> `;
+          })}
+        </select>
+      </div>
+      <input type="hidden" name="__action" value="delete_member" />
+      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${iRow}Popoverdeletemember').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-danger" ${row.users.length > 0 ? '' : 'disabled'}>
+        Delete
+      </button>
+    </form>
+  `;
+}
+
+function uploadAssessmentGroupsModal(csrfToken: string) {
+  return html`
+    <div class="container-fluid">
+      <div
+        class="modal fade"
+        id="uploadAssessmentGroupsModal"
+        tabindex="-1"
+        role="dialog"
+        aria-labelledby="uploadAssessmentGroupsModalLabel"
+      >
```

Comment:
We have a "component" that handles a lot of the Modal boilerplate, you might consider using that here too.

Component: https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/src/components/Modal.html.ts

Example usage: https://github.com/PrairieLearn/PrairieLearn/blob/09e7dce2d6cbbe351d2151b3211616c271d41c0d/apps/prairielearn/src/pages/administratorInstitutions/administratorInstitutions.html.ts#L58